### PR TITLE
Payment total for each transaction + SupaBase field update + refactored previous code

### PR DESCRIPTION
**Pull request by Zea595**
**REQUIRES:**
-transaction has to be a transaction that is **newly created**.

**MODIFIES:**
bullmq -> SupaBase (new column ***"total_amount"*** on database, **floating 8-bit type**)

**EFFECTS:**
-Updates the transaction field ***"total_amount"*** with the expectedTotal on SupaBase
-Shows a total tab on each transactionROW that is the sum of all associated payments

**SUMMARY:**
Basically just added a new column for transactions named ***"total_amount"***. 
***Create.js*** was made to update the database with the ***expectedTotal*** it calculates. 
The total is passed as a prop to be shown on the frontend.

The ***"total_amount"*** of a transaction is only updated during creation *for now*.

**REFACTORED:**
Merged in the previous change regarding expandable list, expand function is now a component object.
Both FEED tables now have a expandable list object that replaces the old table.

### CODE REVIEW

**Comment by Scorpsy** [comment]
I'm gonna need you to show me a picture of the result in your local, what I got from the PR is that on the front end there will be a tab in each transaction for total amount? If that is the case then wouldn't  the amount on the right most table be irrelevant now?

Unless you mean that the total amount would be how much we owe a single person that requested us a series of transaction?

**Comment by Zea595** [comment]
> I'm gonna need you to show me a picture of the result in your local, what I got from the PR is that on the front end there will be a tab in each transaction for total amount? If that is the case then wouldn't the amount on the right most table be irrelevant now?

Here are the pictures to demonstrate the ***total_amount*** functionality
image
As you can see the total amount of this transaction is 23 + 17 which amounts to 40 on the request page.
image
The total is then displayed on each transactionROW' right side on the main ***feed***.

I am not sure what you mean by the "amount on the right most table" but up to the previous point there was only a "{state}" indicator and no numerical value in that position. 
That does remind me that I removed the "{state}" variable by accident, I will make amends to that.
However my PR is meant to add a total number to display, since it tells the user more information about each open transaction.

> 
> Unless you mean that the total amount would be how much we owe a single person that requested us a series of transaction?

No, but I would imagine this would be a nice filter feature of some sort that could be used to sort the **TransactionTable** OR the **InvolvedPaymentTable**. Could be implemented in the future.

File: src/serializers/transactions/transaction-details-serializer.ts [hunk 1]
```diff
@@ -19,6 +20,7 @@ const transactionDetailsSerializer = (transaction: Transaction, user: User): Tra
     createdAt: transaction.created_at,
     state: transaction.state,
     avatarUrl: user.avatar_url,
+    total_amount: transaction.total_amount,
```
Comment (kozr) [inline src/serializers/transactions/transaction-details-serializer.ts:23]: on frontend we use camelCase so lines in the serializer need to be like `totalAmount: transaction.total_amount`

File: src/serializers/transactions/transaction-details-serializer.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export interface TransactionDetails {
   createdAt: Date;
   state: string;
   avatarUrl: string;
+  total_amount: number;
```
Comment (kozr) [inline src/serializers/transactions/transaction-details-serializer.ts:11]: need to cascade the change here too

File: src/pages/feed/components/TransactionRow.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ const TransactionRow = ({ transactionDetails, onClick }: TransactionRowProps) =>
   const authContext = useAuth()
   const currentUser = authContext.user
 
-  const { id, userId, type, name, createdAt, state, avatarUrl } = transactionDetails;
+  const { id, userId, type, name, createdAt, state, avatarUrl, total_amount } = transactionDetails;
```
Comment (kozr) [inline src/pages/feed/components/TransactionRow.tsx:15]: and now this part becomes totalAmount instead

**Review by kozr — APPROVED** [review]
Make the changes then we're good!

**Comment by Zea595** [comment]
> on frontend we use camelCase so lines in the serializer need to be like `totalAmount: transaction.total_amount`

Renamed: 

***total_amount*** to ***totalAmount*** for all front end dependencies 

All changes requested have been made.

**Pull request merged**